From eb96be6e7feec09ac26ea191c3b2a3575b191cd9 Mon Sep 17 00:00:00 2001
From: Pratham Malaviya <prathamm@mechasystems.com>
Date: Mon, 8 Sep 2025 14:50:43 +0530
Subject: [PATCH 1/3] arm64: dts: added usb node and pca9535 IO expander

Add device tree entries for the on-board USB controller and the
PCA9535 I/O expander.  The expander also provides a GPIO used to
drive the haptic motor.

Signed-off-by: Pratham Malaviya <prathamm@mechasystems.com>

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts b/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
index c225316f0ebd..60236cdc387e 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
@@ -7,6 +7,7 @@
 
 #include <dt-bindings/phy/phy-imx8-pcie.h>
 #include <dt-bindings/pwm/pwm.h>
+#include <dt-bindings/usb/pd.h>
 #include "imx8mp.dtsi"
 
 / {
@@ -20,8 +21,7 @@ chosen {
 	/* Reference from imx8mp-evk dts file needs to be tested */
 	memory@40000000 {
 		device_type = "memory";
-		reg = <0x0 0x40000000 0 0xc0000000>,
-		      <0x1 0x00000000 0 0xc0000000>;
+		reg = <0x00000000 0x40000000   0x00000001 0x00000000>;
 	};
 
 	iw612_pwrseq: iw612_pwrseq {
@@ -129,7 +129,32 @@ reg_wifi_en: regulator-wifi-en {
 		 regulator-name = "PDn_AW-CM276NF";
 		 startup-delay-us = <2000>;
 	};
+};
+
+&usb3_0 {
+        status = "okay";
+};
+
+&usb3_phy0 {
+        status = "okay";
+};
+
+&usb_dwc3_0 {
+        status = "okay";
+	dr_mode = "otg";
+};
+
+&usb3_phy1 {
+        status = "okay";
+};
+
+&usb3_1 {
+        status = "okay";
+};
 
+&usb_dwc3_1 {
+        status = "okay";
+	dr_mode = "host";
 };
 
 &irqsteer_hdmi {
@@ -330,14 +355,14 @@ imx219: imx219@10 {
 	        pwn-gpios = <&gpio3 14 GPIO_ACTIVE_HIGH>; //pwn=14
 
 	        port {
-	            imx219_to_mipi_csi2: endpoint {
-	                remote-endpoint = <&mipi_csi_1_in>;
-	                clock-lanes = <0>;
-	                data-lanes = <1 2>;
-	                clock-noncontinuous;
-	                link-frequencies = /bits/ 64 <456000000>;
-	            };
-	       };
+			imx219_to_mipi_csi2: endpoint {
+				remote-endpoint = <&mipi_csi_1_in>;
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				clock-noncontinuous;
+				link-frequencies = /bits/ 64 <456000000>;
+			};
+		};
 	};
 };
 
@@ -413,6 +438,24 @@ &i2c6 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c6>;
 	status = "okay";
+
+	pca9535:gpio@20 {
+	        compatible = "nxp,pca9535";
+	        reg = <0x20>;
+	        gpio-controller;
+	        #gpio-cells = <2>;
+	        interrupt-parent = <&gpio4>;
+	        interrupts = <28 IRQ_TYPE_LEVEL_LOW>;
+	        interrupt-controller;
+	        #interrupt-cells = <2>;
+
+		nSleep {
+	                gpio-hog;
+	                gpios = <15 GPIO_ACTIVE_HIGH>;
+	                output-high;
+	                line-name = "haptic motor nSleep";
+		};
+	};
 };
 
 &uart1 { 
@@ -580,7 +623,6 @@ port {
 			panel_in: endpoint {
 				remote-endpoint = <&dsi_out>;
 			};
-
 		};
 	};
 };
@@ -957,5 +999,4 @@ MX8MP_IOMUXC_SAI1_RXD2__GPIO4_IO04		0x184
 			MX8MP_IOMUXC_SAI1_RXD3__GPIO4_IO05		0x184
 		>;
 	};
-
 };
-- 
2.25.1

